### PR TITLE
[4.0] Convert plugin user terms to prepared statements

### PR DESCRIPTION
**Pull request by HLeithner**
# Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the plugin in all ways you can think of.

Create a form with user terms.

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/user/terms/field/terms.php [hunk 1]
```diff
@@ -79,11 +80,14 @@ protected function getLayoutData()
 
 		if ($termsArticle && Factory::getApplication()->isClient('site'))
 		{
-			$db    = Factory::getDbo();
+			$db           = Factory::getDbo();
+			$termsArticle = (int) $termsArticle;
```
Comment (SharkyKZ) [inline plugins/user/terms/field/terms.php]: Don't need this, already cast as integer above.

Comment (richard67) [reply plugins/user/terms/field/terms.php]: I agree with SharkyKZ, the value assigned to this variable is already casted here, so no need to cast the result: [https://github.com/joomla/joomla-cms/blob/7e3250ad1b7c17a244ad713a06de37d2d02b46a7/plugins/user/terms/field/terms.php#L79](https://github.com/joomla/joomla-cms/blob/7e3250ad1b7c17a244ad713a06de37d2d02b46a7/plugins/user/terms/field/terms.php#L79)

Comment (richard67) [reply plugins/user/terms/field/terms.php]: I agree with SharkyKZ, the value assigned to this variable is already casted here, so no need to cast the result: [https://github.com/joomla/joomla-cms/blob/7e3250ad1b7c17a244ad713a06de37d2d02b46a7/plugins/user/terms/field/terms.php#L79](https://github.com/joomla/joomla-cms/blob/7e3250ad1b7c17a244ad713a06de37d2d02b46a7/plugins/user/terms/field/terms.php#L79)

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 5c826939cafb84bc8bac9d073646d429e0e79d81<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25545">issues.joomla.org/tracker/joomla-cms/25545</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5c826939cafb84bc8bac9d073646d429e0e79d81<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25545">issues.joomla.org/tracker/joomla-cms/25545</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25545">issues.joomla.org/tracker/joomla-cms/25545</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**